pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -77,13 +78,22 @@ func New(context context.Context, changeStore *store.ChangeStore, configStore *s
 	// Get the device capabilities
 	capResponse, capErr := target.CapabilitiesWithString(context, "")
 	if capErr != nil {
-		log.Error(sync.Device.Address, " capabilities ", err)
+		log.Error(sync.Device.Address, " capabilities ", capErr)
 		errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorDeviceCapabilities,
-			string(device.ID), err)
-		return nil, err
+			string(device.ID), capErr)
+		return nil, capErr
 	}
-
-	log.Info(sync.Device.Address, " capabilities ", capResponse)
+	sync.encoding = gnmi.Encoding_PROTO // Default
+	if capResponse != nil { // Currently the case with stratum (Aug 19)
+		for _, enc := range capResponse.SupportedEncodings {
```

Comment:
with this for you are overriding the encoding every time with the latest one. Is that the desired behaviour ? wouldn't' it be better to have a list of supported encoding as sync attribute ?

Reply:
We have to choose one preffered one ourselves at some stage anyway, so I was choosing it here and now. If we keep the list we will have to choose one everytime we call a command.

Reply:
maybe put a todo fix here, we could parametrize this